cfps/CFP-041-clang-format-for-bpf-formatting.md

```diff
@@ -0,0 +1,81 @@
+* Provide make targets for running `clang-format` on the codebase.
+* Disable formatting related checks in checkpatch.pl, leaving linting checks enabled.
+
+## Non-Goals
+
+* Remove checkpatch.pl from the CI pipeline.
+
+## Proposal
+
+### Overview
+
+We propose the following:
+
+* Create a small 3 person committee consisting of Dylan Reimerink and two other people to debate and decide on the formatting rules we want to enforce.
+* Modify the existing `.clang-format` file to reflect the decisions made by the committee.
+* Run `clang-format` on the entire C codebase and commit the changes.
+* Add these changes to a `.git-blame-ignore` file so the commit doesn't mess up the blame history.
+* Change CI to use clang-format to detect formatting issues and switch off formatting checks in checkpatch.pl.
+* Backport these changes to older, still maintained branches to minimize merge conflicts during backporting.
+* Bypass the normal CODEOWNERS review process, instead having the committee review and approve the PR(s) needed for this change.
+
+### The Committee
+
+Even when trying to get clang-format as close to the existing format, there are still a number of decisions that need to be made. Our current formatting is internally inconsistent having multiple standards for indenting, spacing, alignment, etc. between and within files. These aspects where previously not enforced by checkpatch.pl and up to the individual contributor to decide. However, clang-format requires us to make these decisions.
+
+code formatting / code style is infamous for being a topic that can lead to heated debates and bikeshedding. Unfortunately chances are slim that we can make every contributor happy. We propose to create a small committee we trust to make these decisions and to given them the power to make these decisions. The committee will consist of Dylan Reimerink being the committer to work on this primarily and two other people yet to be chosen.
+
+### Existing format and `.clang-format` config
+
+The starting point of this work will be the existing format. We already have a `.clang-format` config file in the repository which has been created in the past to approximate the existing formatting style. The committee will review the changes made by clang-format and decide on the best settings to use.
+
+### Initial re-formatting commit
+
+file so the commit doesn't mess up the blame history.codebase and commit the changes. This will be a single commit to avoid polluting the git history with formatting changes. We will add this commit to a `.git-blame-ignore` which is recognized by both the git cli tool and GitHub to ignore this commit when showing blame information to minimize the impact.
+
+### Tooling changes
+
+We propose to update the cilium-builder docker image which builds our clang to also provide `clang-format`. We will add make targets to run `clang-format` in-place (directly update files) or output any suggested changes to stdout.
```

Comment:
This implicitly standardises on the clang-format contained in cilium-builder, which I think is fine. Maybe it makes sense to call that out though?

Should we also mention that clang-format is updated in tandem with compiler upgrades? (I think these can cause formatting to change again, and might need extra hand holding.)